### PR TITLE
BVC-4754: add validation type custom type get all rules without filtering by ty…

### PR DESCRIPTION
**Pull request by Kurt57**
# Description

 add validation type custom type get all rules without filtering by tpe (ruleType) this is needed for new validation type approach

- add unit test for this case

# Issue Link

# How Has This Been Tested?

- [x] Unit Tests 
- [ ] Snapshot-Test
- [ ] Manual testing (Simulator)
- [ ] Manual testing (Device)

# Screenshots

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)

# Changelog

# Checklist

- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: Sources/CertLogic/CertLogic.swift [hunk 1]
```diff
@@ -15,6 +15,7 @@ public enum ValidationType {
   case issuer
   case destination
   case traveller
+  case custom
```
Comment (ghost) [inline Sources/CertLogic/CertLogic.swift]: Is there really no better name than "custom"? Or add a comment what you mean with it.

File: Sources/CertLogic/CertLogic.swift [hunk 2]
```diff
@@ -140,6 +143,36 @@ final public class CertLogicEngine {
...
+        let rules = groupedCertTypeRulesWith[key]
+        if let maxRules = rules?.max(by: { (ruleOne, ruleTwo) -> Bool in
+           return ruleOne.versionInt < ruleTwo.versionInt
+        }) {
+         returnedRulesItems.append( maxRules)
+        }
+      }
+
+      return returnedRulesItems
+    }
```
Comment (ghost) [inline Sources/CertLogic/CertLogic.swift:175]: It's hard to grasp what you are doing in this function.

File: Sources/CertLogic/CertLogic.swift [hunk 3]
```diff
@@ -140,6 +143,36 @@ final public class CertLogicEngine {
     result = "{" + "\"\(Constants.external)\":" + "\(externalJsonString)" + "," + "\"\(Constants.payload)\":" + "\(payload)"  + "}"
     return result
   }
+
+    // Get List of Rules for Country by Code
+    private func getCustomListOfRulesForAll(filter: FilterParameter, issuerCountryCode: String) -> [Rule] {
+      var returnedRulesItems: [Rule] = []
+      var certTypeRulesWith = rules.filter { rule in
```
Comment (ghost) [inline Sources/CertLogic/CertLogic.swift]: "With" what?

**Review by ghost — APPROVED** [review]

File: Sources/CertLogic/CertLogic.swift (hunk 2 above)
Comment (Kurt57) [reply Sources/CertLogic/CertLogic.swift:175]: its just an copy & paste of some other function to have the method similar to the original authors code

I think in this fork is more important to make minor changes

**Pull request merged**